net_transport.go

```diff
@@ -574,32 +587,46 @@ func (n *NetworkTransport) handleCommand(r *bufio.Reader, dec *codec.Decoder, en
 			isHeartbeat = true
 		}
 
+		labels = []metrics.Label{
+			{
+				Name:  "rpcType",
+				Value: "AppendEntries",
+			},
+			{
```

Comment:
Prometheus doesn't like to see the same metric with different labelsets.  Instead of having the `heartbeat` label present only for this rpcType, can we use two different values - e.g. AppendEntries and AppendEntriesHeartbeat?

Reply:
First, the proposed change would be easy enough to make and I can do that.

For my own knowledge though could you elaborate a bit on what prometheus doesn't like about that and how it impacts using the metrics? I have been using these with prometheus + grafana and haven't run into any strange behavior so far.

Reply:
It's been a while (years) since I ran into issues with this kind of thing, so I'm a little hazy on the details.  Definitely no problem ingesting the data, but some kinds of queries (aggregates maybe?) are more awkward when you have inconsistent labelsets.  Not a huge deal, just not a great pattern IMO.

Reply:
Actually I think it was joins - you have to jump through some extra hoops to ensure the labelsets are consistent or your joins will drop data.  But for sums too it's a bit of a pain, like if a user sees one of the single-label metrics as an example, they might do `sum by (rpcType)` and not realize that there's a meaningful distinction between AppendEntries as a heartbeat and AppendEntries with actual logs, because they got lumped in together.

Reply:
Not that you can really sum quantiles, but the same would apply to `max`.

Reply:
That makes sense. I do know that when crafting some queries utilizing max I did have to add a `{heartbeat="false"}` in the query to get it to ignore those append entries calls. Having a separate type for the heartbeats all together though seems easier to manage.